### PR TITLE
Moving Query Workbench to Dev Tools

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
Screenshot 2024-07-11 at 5 26 49 PM

This change migrates query workbench to dev tools for the navigation changes planned for 2.16. The way the selectedMDSDataConnectionId was passed needed to be updated to allow dev tools to correctly find data sources.

Working with MDS:

https://github.com/user-attachments/assets/d0e02549-c299-40de-b4fd-8de5b15f4d07

Working with Flint:
[FlintTest.mov.zip](https://github.com/user-attachments/files/16185668/FlintTest.mov.zip)

There is an existing issue I outlined in https://github.com/opensearch-project/dashboards-query-workbench/issues/348. But testing with Flint(using dashboards-observability prior to the https://github.com/opensearch-project/dashboards-observability/commit/19d68b2d443458c3fdb1bd0fe035bf823111015b commit) seems to keep all functionality working correctly.

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by SuZhou-Joe — APPROVED** [review]

**Comment by sumukhswamy** [comment]
nit. could you rename all the dataSourceId to dataSourceMDSId, will be easier to distinguish

File: public/application.tsx [hunk 1]
```diff
@@ -14,9 +14,10 @@ import { AppPluginStartDependencies } from './types';
 export const renderApp = (
   { notifications, http, chrome, savedObjects }: CoreStart,
   { navigation, dataSource }: AppPluginStartDependencies,
-  { appBasePath, element, setHeaderActionMenu }: AppMountParameters,
+  { appBasePath, element, setHeaderActionMenu, dataSourceId }: AppMountParameters,
   dataSourceManagement: DataSourceManagementPluginSetup
 ) => {
+  console.log(dataSourceId);
```
Comment (ps48) [inline public/application.tsx:20]: let's remove console.log

**Review by ps48 — COMMENTED** [review]
1. Can we please add the legacy UR support for workbench: https://github.com/opensearch-project/dashboards-observability/blob/7566cff5e9a245cc619732741e05886617134e8e/public/plugin.tsx#L187-L190

   So If user types `https://playground.opensearch.org/app/opensearch-query-workbench` they’ll be redirected to `https://playground.opensearch.org/app/dev_tools#/opensearch-query-workbench`

2. Let's fix the linter errors here. 
Thanks!

**Review by ps48 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

File: public/application.tsx [hunk 2]
```diff
@@ -14,7 +13,7 @@ import { AppPluginStartDependencies } from './types';
 export const renderApp = (
   { notifications, http, chrome, savedObjects }: CoreStart,
   { navigation, dataSource }: AppPluginStartDependencies,
-  { appBasePath, element, setHeaderActionMenu }: AppMountParameters,
+  { appBasePath, element, setHeaderActionMenu, dataSourceId }: AppMountParameters,
```
Comment (ps48) [inline public/application.tsx:16]: @TackAdam Did we test this with and without new nav? cc: @sumukhswamy

Comment (ps48) [reply public/application.tsx:16]: Thanks! Sumukh for validating offline that workbench works with MDS with and without new nav.

**Pull request merged**